hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -459,7 +467,10 @@ private Stream<HoodieInstant> getCommitInstantsToArchive() {
               oldestInstantToRetainForCompaction.map(instantToRetain ->
                       compareTimestamps(s.getTimestamp(), LESSER_THAN, instantToRetain.getTimestamp()))
                   .orElse(true)
-          );
+          ).filter(s ->
```

Comment:
may I know why is this required ?

Reply:
we need to make sure archival does not make progress further than latest clean, so here we compare the instants to not great than the latestCompleteCleanInstant

Reply:
not sure if this is valid for all policies. for eg, wrt Cleaning based on FILE_VERSIONS, I don't think we have any relation b/w archival and cleaning. 
wrt Cleaning based on COMMITS, may be its true. 

So, probably we can't add this enhancement.